### DISCUSSION TITLE
How to map user metadata from different auth providers to profiles table

### DISCUSSION THREAD

**Discussion by rw-simon** [supabase/supabase#11519] [Questions]
Hi there

I set up a register/login system with Email/PW, and with triggers and functions (as example in Supabase SQL Editor) I create a row in the public.profiles table whenever a user registers. I access a username field from the registration form through raw_user_metadata->>'username'.

If I additionally provide OAuth login possibilities (for example Google), where – and how – do I adjust the key of the username field in the user_metadata? In Google it is called name/full_name, and other providers have their own naming also.

Thanks
Simon

**Comment by larrasu** [comment]
Hi, have you figured out a way to do this?
